webapp/components/Map/MapBaseLayerPeriodSelector.js

```diff
@@ -50,44 +100,74 @@ export const MapBaseLayerPeriodSelector = () => {
         ready: true,
         periods: availablePeriods,
         periodByValue: availablePeriods.reduce((acc, period) => ({ ...acc, [getPeriodValue(period)]: period }), {}),
-        selectedPeriodValue: lastPeriodValue,
+        selectedPeriodValueLeft: lastPeriodValue,
+        selectedPeriodValueRight: lastPeriodValue,
+        procLeft: Proc.rgb,
+        procRight: Proc.rgb,
       })
     })()
-  }, [periodSelectorAvailable, provider, periodType])
+  }, [periodSelectorAvailable, provider, periodType, ready])
 
   const onMapLayerPeriodChange = useCallback(
-    (event) => {
+    (isLeft) => (event) => {
       const periodValue = event.target.value
       // replace url in current layer
-      const layer = Object.values(map._layers).find((layer) => layer._url && layer.id === contextBaseLayer.id)
+      let layer
+      if (isLeft) {
+        layer = Object.values(map._layers).find((layer) => layer.options.id !== 'side')
+      } else {
+        layer = Object.values(map._layers).find((layer) => layer.options.id === 'side')
+      }
```

Comment:
this code is repeated maybe this code should go into the Map

Reply:
No reason for this to go to the map when this is especially for this component and is the main thing of this component.